app/src/pages/inside/logsPage/logsGrid/logsGrid.jsx

```diff
@@ -0,0 +1,253 @@
+import { dateFormat } from 'common/utils';
+import { ERROR, FATAL } from 'common/constants/logLevels';
+import ArrowIcon from 'common/img/arrow-down-inline.svg';
+import { NoItemMessage } from '../logItemInfo/logItemInfoTabs/noItemMessage';
+import { LogMessageSearch } from './logMessageSearch';
+import { LogMessageBlock } from './logMessageBlock';
+import { AttachmentBlock } from './attachmentBlock';
+import styles from './logsGrid.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  timeColumnTitle: {
+    id: 'LogsGrid.timeColumnTitle',
+    defaultMessage: 'Time',
+  },
+  noResults: {
+    id: 'LogsGrid.noResults',
+    defaultMessage: 'No results found',
+  },
+});
+const timeColumnId = 'time';
+
+const MessageColumn = ({ className, value, ...rest }) => (
+  <div
+    className={cx('message-column', `level-${value.level}`, className, {
+      console: rest.customProps.consoleView,
+    })}
+    id={value.level === ERROR ? value.id : undefined}
+  >
+    <LogMessageBlock value={value} {...rest} />
+  </div>
+);
+MessageColumn.propTypes = {
+  className: PropTypes.string.isRequired,
+  customProps: PropTypes.object,
+  value: PropTypes.object,
+};
+MessageColumn.defaultProps = {
+  customProps: {},
+  value: {},
+};
+
+const AttachmentColumn = ({ className, value, customProps }) => (
+  <div
+    className={cx('attachment-column', className, {
+      mobile: customProps.mobile,
+      console: customProps.consoleView,
+    })}
+  >
+    {value.binary_content && (
+      <AttachmentBlock customProps={customProps} value={value.binary_content} />
+    )}
+  </div>
+);
+AttachmentColumn.propTypes = {
+  className: PropTypes.string.isRequired,
+  customProps: PropTypes.object,
+  value: PropTypes.object,
+};
+AttachmentColumn.defaultProps = {
+  customProps: {},
+  value: {},
+};
+
+const TimeColumn = ({ className, value, customProps: { mobile } }) => (
+  <div className={cx('time-column', className, { mobile })}>{dateFormat(value.time)}</div>
+);
+TimeColumn.propTypes = {
+  className: PropTypes.string.isRequired,
+  customProps: PropTypes.object,
+  value: PropTypes.object,
+};
+TimeColumn.defaultProps = {
+  customProps: {},
+  value: {},
+};
+
+@injectIntl
+export class LogsGrid extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    logItems: PropTypes.arrayOf(PropTypes.object),
+    loading: PropTypes.bool,
+    filter: PropTypes.string,
+    onFilterChange: PropTypes.func,
+    sortingColumn: PropTypes.string,
+    sortingDirection: PropTypes.string,
+    onChangeSorting: PropTypes.func,
+    consoleView: PropTypes.bool,
+    markdownMode: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    logItems: [],
+    loading: false,
+    filter: '',
+    onFilterChange: () => {},
+    sortingColumn: '',
+    sortingDirection: '',
+    onChangeSorting: () => {},
+    consoleView: false,
+    markdownMode: false,
+  };
+
+  getColumns = () =>
+    this.props.consoleView
```

Comment:
I suggest to split it on 2 separate functions and use them here like this:
```
this.props.consoleView ? this.getConsoleViewColumns() : this.getDefaultViewColumns()
```

Reply:
Done